addons/event_booth_sale/wizard/event_booth_configurator.py

```diff
@@ -13,18 +13,20 @@ class EventBoothConfigurator(models.TransientModel):
     sale_order_line_id = fields.Many2one('sale.order.line', string='Sale Order Line', readonly=True)
     event_id = fields.Many2one('event.event', string='Event', required=True)
     event_booth_category_available_ids = fields.Many2many(related='event_id.event_booth_category_available_ids', readonly=True)
-    event_booth_category_id = fields.Many2one('event.booth.category', string='Booth Category', required=True)
-    event_booth_ids = fields.Many2many('event.booth', string='Booth', required=True)
+    event_booth_category_id = fields.Many2one(
+        'event.booth.category', string='Booth Category', required=True,
+        compute='_compute_event_booth_category_id', readonly=False, store=True)
+    event_booth_ids = fields.Many2many(
+        'event.booth', string='Booth', required=True,
+        compute='_compute_event_booth_ids', readonly=False, store=True)
 
-    @api.onchange('event_id')
-    def _onchange_event_id(self):
-        if not self.env.context.get('default_event_id'):
-            self.event_booth_category_id = False
+    @api.depends('event_id')
+    def _compute_event_booth_category_id(self):
+        self.event_booth_category_id = False
 
-    @api.onchange('event_booth_category_id')
-    def _onchange_event_booth_category_id(self):
-        if not self.env.context.get('default_event_booth_category_id'):
-            self.event_booth_ids = False
+    @api.depends('event_id', 'event_booth_category_id')
+    def _compute_event_booth_ids(self):
```

Comment:
Not sure this computes adds something ? Shouldn't we reset if category changed and booth do not match ?